espflash/src/flasher/mod.rs

```diff
@@ -710,14 +712,19 @@ impl Flasher {
             return Ok(flasher);
         }
 
-        // Load flash stub if enabled
-        if use_stub {
-            info!("Using flash stub");
-            flasher.load_stub()?;
+        if !sdm {
```

Comment:
```suggestion
        if !connection.secure_download_mode {
```
Sorry to nitpick, but I don't think we need to create a local variable which is only referenced in one place.

Reply:
I was about to say that "connection" is moved to "flasher", but then realized that I can use "flasher.connection" actually 🤡 

Will be done, thanks haha 😄